File: defacto-core/src/main/java/org/aksw/defacto/evidence/ComplexProof.java
package org.aksw.defacto.evidence;

import org.aksw.defacto.boa.Pattern;
import org.aksw.defacto.ml.feature.fact.AbstractFactFeatures;
import org.aksw.defacto.model.DefactoModel;

import weka.core.Instance;

/**
 * 
 * @author Daniel Gerber <dgerber@informatik.uni-leipzig.de>
 */
public class ComplexProof {

    private Instance instance = new Instance(AbstractFactFeatures.factFeatures.numAttributes());
    
    private Pattern pattern;
    private WebSite website;
    private DefactoModel model;

    private String firstLabel;
    private String secondLabel;

    private String proofPhrase;
    private String normalizedProofPhrase;

    private double score = 0D;

	private String smallContext;
	private String taggedSmallContext;

	private String mediumContext;
	private String taggedMediumContext;

	private String largeContext;
	private String taggedLargeContext;

	private String tinyContext;
	private String taggedTinyContext;

	private boolean hasPatternInBetween;

    /**
     * boa pattern found
     * 
     * @param firstLabel
     * @param secondLabel
     * @param occurrence
     * @param normalizedOccurrence
     * @param site
     * @param boaPattern
     */
    public ComplexProof(DefactoModel model, String firstLabel, String secondLabel, String occurrence, String normalizedOccurrence, WebSite site, Pattern boaPattern) {
        
        this.model                    = model;
        this.firstLabel               = firstLabel;
        this.secondLabel              = secondLabel;
        this.proofPhrase              = occurrence;
        this.normalizedProofPhrase    = normalizedOccurrence;
        this.website                  = site;
        this.pattern                  = boaPattern;
		this.hasPatternInBetween      = false;
    }

    /**
     * proof with no boa pattern
     * 
     * @param firstLabel
     * @param secondLabel
     * @param occurrence
     * @param normalizedOccurrence
     * @param site
     */

	/*
    public ComplexProof(DefactoModel model, String firstLabel, String secondLabel, String occurrence, String normalizedOccurrence, WebSite site) {
        
        this.model                    = model;
        this.firstLabel               = firstLabel;
        this.secondLabel              = secondLabel;
        this.proofPhrase              = occurrence;
        this.normalizedProofPhrase    = normalizedOccurrence;
        this.website                  = site;
		this.hasPatternInBetween      = false;
    }
	*/

	public void setHasPatternInBetween(boolean value){
		this.hasPatternInBetween = value;
	}

	public boolean getHasPatternInBetween(){
		return this.hasPatternInBetween;
	}

    public String getLanguage(){
    	
    	return this.website.getLanguage();
    }
    
    /**
     * 
     * @return
     */
    public String getProofPhrase() {

        return this.proofPhrase;
    }

    public Instance getFeatures() {

        return instance;
    }

    public WebSite getWebSite() {

        return this.website;
    }

    public Pattern getPattern() {

        return this.pattern;
    }

    public DefactoModel getModel() {

        return this.model;
    }

    public String getNormalizedProofPhrase() {

        return this.normalizedProofPhrase;
    }

    public String getSubject() {

        return this.firstLabel;
    }
    
    public String getObject() {

        return this.secondLabel;
    }

    public void setScore(double score) {

        this.score = score;
    }
    
    public double getScore() {

        return this.score;
    }

	public void setSmallContext(String smallContext) {
		
		this.smallContext = smallContext; 
	}

	public void setMediumContext(String mediumCntext) {
		
		this.mediumContext = mediumCntext;
	}

	public void setLargeContext(String largeContext) {

		this.largeContext = largeContext;
	}

	/**
	 * @return the taggedSmallContext
	 */
	public String getTaggedSmallContext() {
		return taggedSmallContext;
	}

	/**
	 * @param taggedSmallContext the taggedSmallContext to set
	 */
	public void setTaggedSmallContext(String taggedSmallContext) {
		this.taggedSmallContext = taggedSmallContext;
	}

	/**
	 * @return the taggedMediumContext
	 */
	public String getTaggedMediumContext() {
		return taggedMediumContext;
	}

	/**
	 * @param taggedMediumContext the taggedMediumContext to set
	 */
	public void setTaggedMediumContext(String taggedMediumContext) {
		this.taggedMediumContext = taggedMediumContext;
	}

	/**
	 * @return the taggedLargeContext
	 */
	public String getTaggedLargeContext() {
		return taggedLargeContext;
	}

	/**
	 * @param taggedLargeContext the taggedLargeContext to set
	 */
	public void setTaggedLargeContext(String taggedLargeContext) {
		this.taggedLargeContext = taggedLargeContext;
	}

	/**
	 * @return the smallContext
	 */
	public String getSmallContext() {
		return smallContext;
	}

	/**
	 * @return the mediumContext
	 */
	public String getMediumContext() {
		return mediumContext;
	}

	/**
	 * @return the largeContext
	 */
	public String getLargeContext() {
		return largeContext;
	}

	public void setTinyContext(String tinyContext) {
		
		this.tinyContext = tinyContext;
	}
	
	public String getTinyContext() {
		
		return this.tinyContext;
	}

	/**
	 * @return the taggedTinyContext
	 */
	public String getTaggedTinyContext() {
		return taggedTinyContext;
	}

	/**
	 * @param taggedTinyContext the taggedTinyContext to set
	 */
	public void setTaggedTinyContext(String taggedTinyContext) {
		this.taggedTinyContext = taggedTinyContext;
	}

	/* (non-Javadoc)
	 * @see java.lang.Object#hashCode()
	 */
	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((tinyContext == null) ? 0 : tinyContext.hashCode());
		result = prime * result + ((website == null) ? 0 : website.hashCode());
		return result;
	}

	/* (non-Javadoc)
	 * @see java.lang.Object#equals(java.lang.Object)
	 */
	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		ComplexProof other = (ComplexProof) obj;
		if (tinyContext == null) {
			if (other.tinyContext != null)
				return false;
		} else if (!tinyContext.equals(other.tinyContext))
			return false;
		if (website == null) {
			if (other.website != null)
				return false;
		} else if (!website.equals(other.website))
			return false;
		return true;
	}
}

File: python/old/triple/fact_extraction/ComplexProof.py
# -*- coding: utf-8 -*-
"""
@author: JohannesLoevenich

We need to expand this later. Just step by step.
ToDo: Extraction of all features by extract_all_features() method
"""
import NameFeature as nf
import EndOfSentenceFeature as eof

class ComplexProof:
	score = 0
	smallContext = ''
	taggedSmallContext = ''
	mediumContext = ''
	taggedMediumContext = ''
	largeContext = ''
	taggedLargeContext = ''
	#body of the html webpage
	tinyContext = ''
	taggedTinyContext = ''
	def __init__(self,model,firstLabel,secondLabel,occurrence,normalizedOccurence,site):
		self.model = model
		self.firstLabel = firstLabel
		self.secondLabel = secondLabel
		self.proofPhrase = occurrence;
		self.normalizedProofPhrase = normalizedOccurence
		self.website = site
		#Extract EndOfSentenceFeature
	def extract_end_of_sentence_feature(self):
		return eof.EndOfSentenceFeature(self.tinyContext)
		#Extract name feature
	def extract_name_feature(self):
		feature = nf.NameFeature()
		feature.set_values(self.firstLabel,self.proofPhrase,self.secondLabel,self.tinyContext,self.model,self.get_language)
		return feature
	def get_language(self):
		#just en,de
		return self.website.get_language()
	def get_model(self):
		return self.model
	def get_subject(self):
		return self.firstLabel
	def get_object(self):
		return self.secondLabel
	def get_proofPhrase(self):
		return self.proofPhrase
	def get_normalizedProofPhrase(self):
		return self.normalizedProofPhrase
	def get_website(self): 
		return self.website
	 # set methods
	def set_score(self,score):
		self.score = score
	def set_smallContext(self,smallContext):
		self.smallContext = smallContext
	def set_taggedSmallContext(self,taggedSmallContext):
		self.taggedSmallContext = taggedSmallContext
	def set_mediumContext(self,mediumContext):
		self.mediumContext = mediumContext
	def set_taggedMediumContext(self,taggedMediumContext):
		self.taggedMediumContext = taggedMediumContext
	def set_tinyContext(self,tinyContext):
		self.tinyContext = tinyContext
	def set_taggedTinyContext(self,taggedTinyContext):
		self.taggedtinyContext = taggedtinyContext
	 # get methods
	def get_score(self,score):
		return self.score
	def get_smallContext(self,smallContext):
		return self.smallContext 
	def get_taggedSmallContext(self,taggedSmallContext):
		return self.taggedSmallContext
	def get_mediumContext(self,mediumContext):
		return self.mediumContext
	def get_taggedMediumContext(self,taggedMediumContext):
		return self.taggedMediumContext
	def get_tinyContext(self,tinyContext):
		return self.tinyContext
	def get_taggedTinyContext(self,taggedTinyContext):
		return self.taggedtinyContext
